File: research/agent-skills-guard-main/src/types/featured.ts
/**
 * Featured repositories configuration types
 */

export interface FeaturedRepositoriesConfig {
  version: string;
  last_updated: string;
  categories: FeaturedRepositoryCategory[];
}

export interface FeaturedRepositoryCategory {
  id: string;
  name: {
    en: string;
    zh: string;
  };
  description: {
    en: string;
    zh: string;
  };
  repositories: FeaturedRepository[];
}

export interface FeaturedRepository {
  url: string;
  name: string;
  description: {
    en: string;
    zh: string;
  };
  tags: string[];
  featured: boolean; // Whether to feature on the homepage
}

File: research/agent-skills-guard-main/src-tauri/src/models/featured.rs
use serde::{Deserialize, Serialize};
use std::collections::HashMap;

/// Featured repositories configuration
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct FeaturedRepositoriesConfig {
    pub version: String,
    pub last_updated: String,
    pub categories: Vec<FeaturedRepositoryCategory>,
}

/// A category of featured repositories
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct FeaturedRepositoryCategory {
    pub id: String,
    pub name: HashMap<String, String>,
    pub description: HashMap<String, String>,
    pub repositories: Vec<FeaturedRepository>,
}

/// A featured repository entry
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct FeaturedRepository {
    pub url: String,
    pub name: String,
    pub description: HashMap<String, String>,
    pub tags: Vec<String>,
    pub featured: bool,
}
